data-protocols/ids/ids-token-validation/src/main/java/org/eclipse/dataspaceconnector/ids/token/validation/rule/IdsValidationRule.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright (c) 2022 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.token.validation.rule;
+
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.dataspaceconnector.iam.oauth2.spi.ValidationRule;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.ParseException;
+import java.util.Map;
+
+public class IdsValidationRule implements ValidationRule {
+    private final boolean validateReferring;
+
+    public IdsValidationRule(boolean validateReferring) {
+        this.validateReferring = validateReferring;
+    }
+
+    /**
+     * Validates the JWT by checking extended IDS rules.
+     */
+    @Override
+    public Result<SignedJWT> checkRule(SignedJWT jwt, @Nullable Map<String, Object> additional) {
+        if (additional != null) {
+            var issuerConnector = additional.get("issuerConnector");
+            if (issuerConnector == null) {
+                return Result.failure("Required issuerConnector is missing in message");
+            }
+
+            String securityProfile = null;
+            if (additional.containsKey("securityProfile")) {
+                securityProfile = additional.get("securityProfile").toString();
+            }
+
+            return verifyTokenIds(jwt, issuerConnector.toString(), securityProfile);
+
+        } else {
+            throw new EdcException("Missing required additional information for IDS token validation");
+        }
+    }
+
+    private Result<SignedJWT> verifyTokenIds(SignedJWT jwt, String issuerConnector, @Nullable String securityProfile) {
+        try {
+            var claims = jwt.getJWTClaimsSet().getClaims();
+
+            //referringConnector (DAT, optional) vs issuerConnector (Message-Header, mandatory)
+            var referringConnector = claims.get("referringConnector");
+
+            if (Boolean.TRUE.equals(validateReferring) && referringConnector != null && !referringConnector.equals(issuerConnector)) {
```

Comment:
the condition can be simplified in 
`validateReferring && !issuerConnector.equals(referringConnector)`

Reply:
Done (please check again if ok)